src/global/a11y/interactivity-checker.ts

```diff
@@ -0,0 +1,38 @@
+export function hasGeometry(element: HTMLElement): boolean {
+  // Use logic from jQuery to check for an invisible element.
+  // See https://github.com/jquery/jquery/blob/master/src/css/hiddenVisibleSelectors.js#L12
+  return !!(
+    element.offsetWidth ||
+    element.offsetHeight ||
+    (typeof element.getClientRects === 'function' && element.getClientRects().length)
+  );
+}
+
+export interface InterfaceInteractivityChecker {
+  isVisible(element: HTMLElement): boolean;
+}
+
+export class InteractivityChecker implements InterfaceInteractivityChecker {
+  /**
+   * Gets whether an element is visible for the purposes of interactivity.
+   *
+   * This will capture states like `display: none` and `visibility: hidden`, but not things like
+   * being clipped by an `overflow: hidden` parent or being outside the viewport.
+   *
+   * @returns Whether the element is visible.
+   */
+  public isVisible(element: HTMLElement): boolean {
+    return hasGeometry(element) && getComputedStyle(element).visibility === 'visible';
+  }
+}
+
+class InteractivityCheckerMock implements InterfaceInteractivityChecker {
+  public isVisible(): boolean {
+    return true;
+  }
+}
+
+export const interactivityChecker: InterfaceInteractivityChecker =
+  getComputedStyle(document.documentElement).visibility === ''
+    ? new InteractivityCheckerMock()
+    : new InteractivityChecker();
```

Comment:
No tests on this?

Reply:
It's hard to have a proper element in the test environment (with an actual visibility and width/height) and mocking all the values doesn't really make sense.